crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -40,10 +40,18 @@ impl<'interner> TypeChecker<'interner> {
 
                         let first_elem_type = elem_types.get(0).cloned().unwrap_or(Type::Error);
 
-                        let arr_type = Type::Array(
-                            Box::new(Type::Constant(arr.len() as u64)),
-                            Box::new(first_elem_type.clone()),
-                        );
+                        let arr_type = if arr.is_empty() {
+                            dbg!("arr.is_empty()");
+                            Type::Array(
+                                Box::new(Type::Constant(0u64)),
+                                Box::new(self.interner.next_type_variable()),
+                            )
+                        } else {
+                            Type::Array(
+                                Box::new(Type::Constant(arr.len() as u64)),
+                                Box::new(first_elem_type.clone()),
+                            )
+                        };
```

Comment:
The dbg and the whole `if` statement can be removed.

Instead of defaulting `first_elem_type` to `Type::Error` if there is no first element, it should be `.unwrap_or_else(|| next type variable)`

Reply:
Nice much cleaner I switched it